src/main/scala/io/iohk/ethereum/domain/BlockBody.scala

```diff
@@ -3,19 +3,136 @@ package io.iohk.ethereum.domain
 import io.iohk.ethereum.domain.BlockHeaderImplicits._
 import io.iohk.ethereum.rlp.{RLPEncodeable, RLPList, RLPSerializable, rawDecode}
 
-case class BlockBody(transactionList: Seq[SignedTransaction], uncleNodesList: Seq[BlockHeader]) {
-  override def toString: String =
-    s"BlockBody{ transactionList: $transactionList, uncleNodesList: $uncleNodesList }"
+import scala.collection.immutable
+
+class BlockBody private(
+                        private val transactionList: Array[SignedTransaction],
+                        val uncleNodesList: Seq[BlockHeader]
+                      ) extends Iterable[SignedTransaction] {
+
 
-  lazy val numberOfTxs: Int = transactionList.size
+  lazy val numberOfTxs: Int = transactionList.length
 
   lazy val numberOfUncles: Int = uncleNodesList.size
+
+  /**
+    * Works as `copy` for case classes
+    * @param transactions - new transactions for the block
+    *        Beware using array-based collections as an argument, because:
+    *  {{{
+    *  val ar1 = Array(1, 2, 3)
+    *  val seq = arr.toSeq
+    *  val ar2 = seq.toArray
+    *  ar2.update(0,0)
+    *  ar1
+    *  }}}
+    *  would return (0, 2, 3), because ar1 === ar2
```

Comment:
you suggest that `ar1 === ar2` after ar2.update(0,0) ? i mean, AFAIK that is NOT true..

Reply:
@mirkoAlic thanks! I just wanted to mention that `ar1` is the same instance as `ar2` here.
Hence whenever you modify the first one - the second is also updated.
`.toSeq` just wraps an `Array` by producing a `WrappedArray` instance.
And `WrappedArray.toArray` call returns the array it wraps back.